Automattic/vip-go-mu-plugins
# Search: Allow no active versions and change initial state to no versions

## Description
This change does the below:
- Allows for a state of having no active versions in the index version management (before, we would default to 1 no matter what)
- Adds a `deactivate` WP-CLI command: `wp vip-search index-versions deactivate post 1`
- Defaults to having no index versions on initial state

If there's no active versions but search is enabled, search will fall back to DB.

## Changelog Description

### Plugin Updated: Enterprise Search

Add deactivate CLI command, allow a state of having no active indexes and set initial state to no versions

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
Note: when testing on dev-env, you'll need to run `wp vip-search index --setup` since the current version uses `wp elasticsearch index --setup` which doesn't take into account the changes in this PR. see https://github.com/Automattic/vip-container-images/pull/390
1) Run `wp vip-search index-versions deactivate post <index-id>` on an active version and use `wp vip-search index-versions list post` to verify it's no longer marked as active
2) Run a search query via `?s=<searchTerm>` and observe that in Search Dev Tools, there's no query run anymore
3) Run `wp vip-search index-versions delete post <index-id` until there's no more indexes left in `wp vip-search index-versions list post`
4) Run `wp vip-search index --using-versions --setup --indexables=post` and expect an error message returned
5) Run`wp vip-search index --setup` and verify that a new version has been created per `wp vip-search index-versions list post`
6) Repeat step 5 but with the expectation that the is only one index and it's active and different from the one in step 6)
7) Run`wp vip-search index --setup --indexables=post --version=<index id from step 6>` and verify that no new indexes have been added and only the index from step 6 was indexed
